### DISCUSSION TITLE
Accessing data.details from error message in JS SDK

### DISCUSSION THREAD

**Discussion by vesamet** [pocketbase/pocketbase#5155] [Q&A]
In the error responses returned to the JS SDK when doing any request, the property `e.data.details` is `undefined`.
Yet, this data is accessible from Pocketbase's log visual interface ( `localhost:8080/_/#/logs`) 

Ex: If we run the above code, details is undefined. 
```js
  try {
    const authData = await pb
      .collection("users")
      .authWithPassword("user", "wrong-password")
  } catch (err) {
     console.log(err.message) // Failed to authenticate.
     console.log(err.data.details) // undefined
  }
```

If someone could enlighten me on a way to enable it (or use onBeforeApiError() hook to inject it in the response).

**Comment by ganigeorgiev** [comment]
There is no such field as `data.details` in the error response. All JS SDK request errors are wrapped in a `ClientResponseError` exception. For more details check https://github.com/pocketbase/js-sdk?tab=readme-ov-file#error-handling.

If you are trying to access the detailed error message that is printed in the logs, then this is intentionally not returned to the user as part of the response as it could contain sensitive information and could be a security issue.

It is possible to write a custom error response but without details why do you need it I can't recommend it.

**Reply by aarongeorge** [reply]
It's useful to have as currently if you try to login all you get is "Failed to authenticate" and no data about if it's the username or the password that was the issue. Sure you can argue that's a security issue, but what about registration. If your confirm password isn't the same as your password you get back "Failed to create record", yet the logs tell you why it failed (passwordConfirm: Values don't match..). That's a legitimately useful piece of information to return.

Is it possible to explain how we could enable/disable this additional details response (previously PocketBase returned it). Off by default is fine, but being able to get more accurate reasoning from the server as to why a request failed is very helpful (especially when all the PB interactions are handled by the server and not the client)

Thanks :)

**Reply by ganigeorgiev** [reply]
> (previously PocketBase returned it)

That's not true. PocketBase has never returned the raw errors. Only field validation errors are returned to the public APIs. Specifically for the password login endpoint the error message was always generic when a user with the specified email didn't exist or the provided password was incorrect. You will get "data.email" or "data.password" field errors if the provided email is invalid formatted or the password is missing from the body because there is no risk of information disclosure here.

In any case, there are no plans to change the current behavior because again the raw errors could contain sensitive information, including OS paths, database errors or some other internal specifics.

The error API response is standardized for all endpoints and has the following format:

```js
{
  "code": 400,
  "message": "Failed to create record.",
  "data": {
    "title": {
      "code": "validation_required",
      "message": "Missing required value."
    }
  }
}

// to access it from the JS SDK:
try {
    ...
} catch (err) {
    console.log(err.response.data.title)
}
```

**Reply by aarongeorge** [reply]
Okay that's my mistake. I had read (while looking into how to do this) that it previously was possible to view the data.details value, that seems to have been incorrect information.

I completely understand having safe defaults, but I also feel like returning that the passwords didn't match is also a sane default to return over an uninformative message for an end user such as "Failed to create record".

I was more asking if there could be a flag or for more accurate errors to come back in certain situations (as above) but if that's not the case, I'll fork and add myself
